scripts/scr_get_item_names/scr_get_item_names.gml

```diff
@@ -0,0 +1,806 @@
+//  so we prefix them with an underscore
+global.slot_names = {
+    "_1": { // Infantry
+        "_1": "Left Hand",
+        "_2": "Right Hand",
+        "_3": "Armour",
+        "_4": "Gear",
+        "_5": "Mobility"
+    },
+    "_6": { // Dreadnought
+        "_1": "Left Arm",
+        "_2": "Right Arm",
+        "_5": "Accessory"
+    },
+    "_50": { // Land Raider
+        "_1": "Front",
+        "_2": "Sponson",
+        "_3": "Pintle",
+        "_4": "Upgrade",
+        "_5": "Accessory"
+    },
+    "_51": { // Rhino
+        "_1": "Weapon",
+        "_4": "Upgrade",
+        "_5": "Accessory"
+    },
+    "_52": { // Predator
+        "_1": "Turret",
+        "_2": "Sponsons",
+        "_3": "Pintle",
+        "_4": "Upgrade",
+        "_5": "Accessory"
+    },
+    "_53": { // Land Speeder
+        "_1": "Primary",
+        "_2": "Secondary"
+    },
+    "_54": { // Whirlwind
+        "_1": "Missiles",
+        "_2": "Pintle",
+        "_4": "Upgrade",
+        "_5": "Accessory"
+    }
+};
+
+/// @description This function returns the name of the slot for a given unit type and equipment type.
+/// @param {real} _unit_type - The type of unit to equip, see eUNIT_TYPE.
+/// @param {real} _equipment_type - The type of equipment to equip, see eEQUIPMENT_TYPE.
+/// @returns {string} The name of the slot.
+function get_slot_name(_unit_type, _equipment_type) {
+    var _unit_key_str = $"_{_unit_type}";
+    var _equip_key_str = $"_{_equipment_type}";
+
+    if (global.slot_names[_unit_key_str] && global.slot_names[_unit_key_str][_equip_key_str]) {
+        return global.slot_names[_unit_key_str][_equip_key_str];
+    }
+    return "Unknown";
+}
+
+/// @description This function is used to populate the weapon/equipment selection list in the equipment screen.
+/// @param {real} _equipment_type - The type of equipment to equip, see eEQUIPMENT_TYPE.
+/// @param {real} _equipment_subtype - The subtype of equipment to equip, see eEQUIPMENT_SUBTYPE.
+/// @param {real} _unit_type - The type of unit to equip, see eUNIT_TYPE.
+/// @param {bool} _include_company_standard - Whether to include the Company Standard in the selection list.
+/// @param {bool} _show_available_only - Whether to limit the selection to what is in inventory, or show all items.
+/// @param {bool} _master_crafted_only - Whether to show only master crafted items, or hide them.
+/// @returns {array} The list of items to populate the selection list with.
+function scr_get_item_names(_equipment_type, _equipment_subtype, _unit_type, _include_company_standard=false, _show_available_only=false, _master_crafted_only=false) {
+    show_debug_message("scr_get_item_names: " + string(_equipment_type) + ", " + string(_equipment_subtype) + ", " + string(_unit_type) + ", " + string(_include_company_standard) + ", " + string(_show_available_only) + ", " + string(_master_crafted_only));
+    var _item_names;
+    switch(_unit_type) {
```

Comment:
I think you may want to do this, otherwise there is a crash if you click on a slot that should have no items, caused by this line:
`show_debug_message("scr_get_item_names: " + string(_item_names));`
```suggestion
    var _item_names = [];
```

Reply:
Not sure why GitHub didn't mark this as outdated, but I've removed the local _item_names and the debug message, so this isn't an issue in the current PR.